DEVELOPMENT_GUIDE.md

```diff
@@ -0,0 +1,50 @@
+# Valkey development guidelines
+This document provides an general overview for writing and designing code for Valkey.
+During our long development history, we've made a lot of inconsistent decisions, but we strive to get incrementally better.
+
+## General style guidelines
+Most of the style guidelines are enforced by clang format, but some additional comments are included here.
+
+1. C style comments `/* comment */` can be used for both single and multi-line comments. C++ comments `//` can only be used for single line comments.
+1. Generally keep line lengths below 90 characters, however there is no explicit line length enforcement.
```

Comment:
I believe we landed on 120'ish previously but I am personally more in favor of 90 .